cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -1208,6 +1215,15 @@ TEST(Cast, TimeZeroCopy) {
                     time64(TimeUnit::MICRO));
 }
 
+TEST(Cast, DateToString) {
+  for (auto string_type : {utf8(), large_utf8()}) {
+    CheckCast(ArrayFromJSON(date32(), "[0, null]"),
+              ArrayFromJSON(string_type, R"(["1970-01-01", null])"));
+    CheckCast(ArrayFromJSON(date64(), "[86400000, null]"),
+              ArrayFromJSON(string_type, R"(["1970-01-02", null])"));
+  }
+}
+
```

Comment:
Since we're adding timestamp support, can we also have a test like this for timestamps? There's other tests for timestamps around L1066.